lib/default.nix

```diff
@@ -7,6 +7,9 @@ let
 
   inherit (import ./fixed-points.nix {}) makeExtensible;
 
+  # Allow extending lib by specifying an overlay file.
+  finalLib = lib.try (lib.fix (lib.extends (import <lib-overlay>) (self: lib))) lib;
```

Comment:
This defeats the split between for example `pkgs/top-level/impure.nix` and `pkgs/top-level/default.nix`.
Pinning Nix expressions only works because Nixpkgs (a) provides pure versions of things or (b) lets the caller set the argument instead of an impure default. This not only breaks (a), but also breaks (b).